### PR TITLE
glance: Avoid greenthreads hanging endlessly on "slowloris" attack

### PR DESCRIPTION
**Pull request by toabctl**
See https://bugs.launchpad.net/nova/+bug/1361360 for the full
story.

### CODE REVIEW

File: chef/cookbooks/glance/templates/default/glance-registry.conf.erb [hunk 1]
```diff
@@ -171,6 +171,7 @@ workers = <%= [node["cpu"]["total"], 2, 4].sort[1] %>
 # have to set this option to False when you create a wsgi server.
 # (boolean value)
 #http_keepalive = true
+http_keepalive = false
```
Comment (aplanas) [inline chef/cookbooks/glance/templates/default/glance-registry.conf.erb]: Is this needed? As I can see in the link the change is only in glance-api.conf

Comment (toabctl) [inline chef/cookbooks/glance/templates/default/glance-registry.conf.erb]: It's also there for the registry service. See https://review.openstack.org/#/c/130839/ (linked in the bug). Or am I missing something?

Comment (aplanas) [inline chef/cookbooks/glance/templates/default/glance-registry.conf.erb]: My bad, I checked the comments but not the patch.

**Comment by aplanas** [comment]
+1

**Comment by vuntz** [comment]
+1.

Do you have a list of other components where we still need the change?

**Comment by toabctl** [comment]
> Do you have a list of other components where we still need the change?

I think @dirkmueller did already the other services in bb7420313010995c0ba62bad70e9ecb4d99df04a . Manila is also already fixed in 2259327999453121a395edeae1fe23f76590c40d . So I think glance is the only missing service.

**Comment by dirkmueller** [comment]
+1

**Pull request merged**